pandas/core/arrays/integer.py

```diff
@@ -293,6 +294,33 @@ def __array__(self, dtype=None):
         """
         return self._coerce_to_ndarray()
 
+    _HANDLED_TYPES = (np.ndarray, numbers.Number)
+
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+
+        out = kwargs.get('out', ())
+
+        for x in inputs + out:
+            if not isinstance(x, self._HANDLED_TYPES + (IntegerArray,)):
+                return NotImplemented
+
+        if method == '__call__':
+            if ufunc.signature is None and ufunc.nout == 1:
+                args = [a._data for a in inputs]
+                masks = [a._mask for a in inputs]
+                result = ufunc(*args, **kwargs)
+                mask = np.logical_or.reduce(masks)
+                if result.dtype.kind in ('i', 'u'):
+                    return IntegerArray(result, mask)
+                else:
+                    result[mask] = np.nan
+                    return result
+
```

Comment:
We should be able to handle binary ufuncs like `np.add(IntegerArray, 2)` right? Do you want to do that here?

Reply:
Ah, I see you have a test for this.

Reply:
Yes, for sure we should handle that. I didn't have any test for that yet, only for `np.add(IntegerArray, IntegerArray)`

Basically what I did up to now was a basic implemention on IntegerArray, just to be able to test the machinery on Series as well.

Reply:
This is handled by the `maybe_dispatch_ufunc_to_dunder_op` now